src/extractor/guidance/turn_handler.cpp

```diff
@@ -31,6 +51,67 @@ bool isEndOfRoad(const ConnectedRoad &,
            angularDeviation(possible_right_turn.angle, possible_left_turn.angle) >
                2 * NARROW_TURN_ANGLE;
 }
+
+template <typename Iterator>
+Iterator findOutermostForkCandidate(const Iterator start, const Iterator end)
+{
+    const auto outermost = std::adjacent_find(start, end, isOutermostForkCandidate);
+    if (outermost != end)
+    {
+        return outermost;
+    }
+    // if all roads are part of a fork, set `candidate` to the last road
+    else
+    {
+        return outermost - 1;
+    }
+}
+
+struct StraightestTurnAtIntersection
```

Comment:
adding a dedicated struct here seems wrong to me. the straightmost turn at an instersection can be found with `getClosestTurn(STRAIGHT_ANGLE)` for any intersection and is an iterator into the intersection. It's deviation can easily be computed whenever necessary by `angularDeviation(itr.angle,STRAIGHT_ANGLE)`.

The dedicated struct is overkill in my personal opinion. Is it used often enough to justify the definition?